march_gain_scheduling/src/march_gain_scheduling/dynamic_pid_reconfigurer.py

```diff
@@ -27,34 +27,33 @@ def gait_selection_callback(self, data):
                 gait_type=new_gait_type))
             new_gait_type = 'walk_like'
 
-        if self._gait_type != new_gait_type or not self.interpolation_done:
-            self._gait_type = new_gait_type
-            self.load_current_gains()
-            self.interpolation_done = False
+        self._gait_type = new_gait_type
+        self.load_current_gains()
+        needed_gains = [self.look_up_table(i) for i in range(len(self._joint_list))]
+        rospy.logwarn(needed_gains)
+        self.done_interpolation_test(needed_gains)
 
-            needed_gains = [self.look_up_table(i) for i in range(len(self._joint_list))]
+        if not self.interpolation_done:
```

Comment:
self.interpolation_done no longer needs to be a class variable, as it is only used within this method.
```suggestion
        if not interpolation_done:
```